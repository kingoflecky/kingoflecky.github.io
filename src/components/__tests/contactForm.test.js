import React from 'react';
import { screen, render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import '@testing-library/jest-dom';

import ContactForm from "../contactForm";

describe("form fields", () => {
  it("should match snapshot", () => {
    const { asFragment } = render(<ContactForm />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should contain submit button", () => {
    render(<ContactForm />);
    expect(screen.getByRole('button')).toBeInTheDocument();
  });
});
