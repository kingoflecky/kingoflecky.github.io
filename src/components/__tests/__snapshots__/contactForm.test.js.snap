// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`form fields should match snapshot 1`] = `
<DocumentFragment>
  <form
    action="https://public.herotofu.com/v1/7400a8d0-09cd-11ee-8267-d3eb100789b4"
    method="POST"
  >
    <div
      id="formFieldsWrapper"
    >
      <div
        class="formFields"
      >
        <label
          class="formFieldNames"
          for="name"
        >
          Name
        </label>
        <input
          id="name"
          name="name"
          required=""
          type="text"
          value=""
        />
      </div>
      <div
        class="formFields"
      >
        <label
          class="formFieldNames"
          for="email"
        >
          Email Address
        </label>
        <input
          id="email"
          name="email"
          required=""
          type="email"
          value=""
        />
      </div>
      <div
        class="formFields"
      >
        <label
          class="formFieldNames"
          for="subject"
        >
          Subject
        </label>
        <input
          id="subject"
          name="subject"
          required=""
          type="text"
          value=""
        />
      </div>
      <div
        class="formFields"
      >
        <label
          class="formFieldNames"
          for="message"
        >
          Message
        </label>
        <textarea
          id="message"
          name="message"
          required=""
        />
      </div>
      <button
        class="submitBtn"
        id="submit"
        name="submit"
        type="submit"
        value="submit"
      >
         submit
      </button>
    </div>
  </form>
</DocumentFragment>
`;
